system/handlers/admin/systemAlerts/invalidRuleEngineRules.cfc

```diff
@@ -0,0 +1,51 @@
+component {
+	property name="validationEngine"    inject="ValidationEngine";
+	property name="conditionService"    inject="RulesEngineConditionService";
+	property name="systemAlertsService" inject="SystemAlertsService";
+
+	private void function runCheck( required systemAlertCheck check ) {
+		var invalidRules  = [];
+		var ruleFilter    = {};
+		var existingAlert = systemAlertsService.getAlert( type="invalidRuleEngineRules" );
+
+		if ( arrayLen( existingAlert.data.invalidRules ?: [] ) ) {
+			ruleFilter.id = existingAlert.data.invalidRules;
+		}
+
+		var allRules = getPresideObject( "rules_engine_condition" ).selectData( filter=ruleFilter, selectFields=[ "id", "context", "filter_object", "expressions" ] );
```

Comment:
Not sure about the approach here. If an alert has been raised, then re-running the check will only ever recheck the previously found errors, and not find any new ones.

If we were to use the "references()" functionality to get all condition IDs:
* calling runCheck() with no reference would then get all references and run each in turn
* calling runCheck() with a reference would run the check for a specific condition, a more efficient way when adding/editing/deleting conditions
* we would have a separate alert for each failing condition

Reply:
Yes, the point is that the only time a condition/filter will be expected to be invalid is a new version of the software that means missing fields/expressions, etc.

So: on startup, we'd want to scan all
During running: check individual ones after they're edited (and possibly when inserted)